tutorials/desktop/example-mir-desktop/example-mir-desktop.md

```diff
@@ -0,0 +1,547 @@
+    {
+        switch (mir_keyboard_event_scan_code(event))
+        {
+            case KEY_F4:
+                tools.ask_client_to_close(tools.active_window());
+                return true;
+
+            case KEY_TAB:
+                tools.focus_next_application();
+                return true;
+
+            case KEY_GRAVE:
+                tools.focus_next_within_application();
+                return true;
+
+            default:;
+        }
+    }
+
+    return false;
+}
+```
+
+### Keeping score
+
+There's very little code needed to get this basic shell running:
+
+```bash
+$ wc -l *.h *.cpp *.sh
+   88 egwindowmanager.h
+   34 egmde.cpp
+  420 egwindowmanager.cpp
+   47 egmde-desktop.sh
+  589 total
+```
+
+## Step 2: keymap and wallpaper
+Duration: 4:00
+
+At the end of step 1 we could run egmde as a desktop and run and use Wayland based applications. Those of us in Europe (or elsewhere outside the USA) will soon notice that the keyboard layout has defaulted to US, so I’ll show how to fix that. And the black background is rather depressing, so I’ll show how to implement a simple wallpaper; and, finally, how to allow the user to customize the wallpaper.
+
+Along the way we’ll discuss the way the Mir API works with us to make it easy to combine features.
+
+Assuming you're still in the `build` directory used in step 1.
+
+```bash
+git checkout article-2
+make
+```
+After this you can once again start a basic egmde based desktop:
+
+```bash
+./egmde-desktop
+```
+![step-2](images/article-2.png)
+
+You should see a simple gradient wallpaper with a weston-terminal session.
+
+You can set the wallpaper colour when starting the desktop by supplying a --wallpaper parameter, and the keyboard layout with --keymap like this:
+
+```bash
+./egmde --wallpaper 0xff3737 --keymap gb
+```
+
+Alternatively, these options can be specified in a config file:
+
+```bash
+$ cat ~/.config/egmde.config 
+wallpaper=0xff0d0
+keymap=gb
+```
+
+## Step 2: The `main()` code
+Duration: 3:00
+
+The previous code is largely unchanged. Just the egmde.cpp file is changed to add a couple of new headers and update the main program that looked like this:
+
+```cpp
+int main(int argc, char const* argv[])
+{
+    MirRunner runner{argc, argv};
+
+    return runner.run_with(
+        {
+            set_window_management_policy<egmde::WindowManagerPolicy>()
+        });
+}
+```
+
+Now it is:
+
+```cxx
+int main(int argc, char const* argv[])
+{
+    MirRunner runner{argc, argv};
+
+    egmde::Wallpaper wallpaper;
+
+    runner.add_stop_callback([&] { wallpaper.stop(); });
+
+    return runner.run_with(
+        {
+            Keymap{},
+            CommandLineOption{
+                std::ref(wallpaper), 
+                "wallpaper", 
+                "Colour of wallpaper RGB", 
+                "0x92006a"},
+            StartupInternalClient{"wallpaper", std::ref(wallpaper)},
+            set_window_management_policy<egmde::WindowManagerPolicy>()
+        });
+}
+```
```

Comment:
Does markdown let you highlight the lines of code that you added?

Reply:
Not that I can find

Reply:
pity